packages/pwa-kit-react-sdk/src/ssr/server/react-rendering.js

```diff
@@ -230,27 +236,38 @@ export const render = async (req, res, next) => {
     const {html, routerContext, error} = renderResult
     const redirectUrl = routerContext.url
     const status = (error && error.status) || res.statusCode
-
     res.__performanceTimer.mark(PERFORMANCE_MARKS.renderToString, 'end')
     res.__performanceTimer.mark(PERFORMANCE_MARKS.total, 'end')
     res.__performanceTimer.log()
 
-    if (includeServerTimingHeader) {
+    if (shouldTrackPerformance(req)) {
         res.setHeader('Server-Timing', res.__performanceTimer.buildServerTimingHeader())
-
         // Override cache-control header to no caching when __server_timing is used
         // This happens after React rendering is complete, ensuring it overrides any
         // cache headers set by individual page components
         res.set('Cache-Control', NO_CACHE)
     }
 
+    // Cleanup performance timer and OpenTelemetry tracing after response is sent
+    res.__performanceTimer.cleanup()
+    shutdownServerTracing()
+
     if (redirectUrl) {
         res.redirect(routerContext.status || 302, redirectUrl)
     } else {
         res.status(status).send(html)
     }
 }
 
+export const render = (req, res, next) => {
+    res.__performanceTimer = new PerformanceTimer({enabled: shouldTrackPerformance(req)})
+    if (shouldTrackPerformance) {
```

Comment:
Are we missing passing `req` as a parameter?

```suggestion
    if (shouldTrackPerformance(req)) {
```